api/resources_portal/views/auth.py

```diff
@@ -0,0 +1,119 @@
+from django.conf import settings
+from django.http import JsonResponse
+from rest_framework import viewsets
+
+import orcid
+import requests
+from django_expiring_token.authentication import token_expire_handler
+from django_expiring_token.models import ExpiringToken
+
+from resources_portal.config.logging import get_and_configure_logger
+from resources_portal.models.grant import Grant
+from resources_portal.models.organization import Organization
+from resources_portal.models.user import User
+
+logger = get_and_configure_logger(__name__)
+
+CLIENT_ID = settings.CLIENT_ID
+CLIENT_SECRET = settings.CLIENT_SECRET
+OAUTH_URL = settings.OAUTH_URL
+
+
+def remove_code_parameter_from_uri(url):
+    """
+    This removes the "code" parameter added by the first ORCID call if it is there,
+     and trims off the trailing '/?' if it is there.
+    """
+    return url.split("code")[0].strip("&").strip("/?")
+
+
+class AuthViewSet(viewsets.ViewSet):
+
+    # Verify that this should be a GET/POST?
```

Comment:
What's this comment about?

Reply:
I was a bit worried about security on this endpoint, but we've now verified that nothing protected is being passed here. Removed.